app/helpers/samples_helper.rb

```diff
@@ -196,6 +215,40 @@ def get_total_runtime(pipeline_run, run_stages)
     end
   end
 
+  def filter_by_time(samples, start_date, end_date)
```

Comment:
make all the `filter_by_` methods private ?

Reply:
Nothing wrong with that. But that could apply to most methods on this file.

Reply:
I'm of the school that methods should be private if they are not intended to be called outside of the class. Minimize the size of the interface.

Reply:
I guess I do not see a problem to force developers to go through the aggregated  `filter_samples` since it is pretty configurable.